src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/client/HomeOfficeCaseStatusClient.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.infrastructure.client;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientResponseException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCaseDefinition;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.infrastructure.client.util.HomeOfficeApiUtil;
+
+@Service
+public class HomeOfficeCaseStatusClient {
+
+    private final RestTemplate restTemplate;
+    private final String homeOfficeSearchByParametersUrl;
+    private HomeOfficeApiUtil homeOfficeApiUtil;
+
+    public HomeOfficeCaseStatusClient(
+        @Value("${homeoffice.endpoint}") String homeOfficeEndpoint,
+        @Value("${homeoffice.case.search.uri}") String homeOfficeSearchByParametersUrl,
+        RestTemplate restTemplate,
+        HomeOfficeApiUtil homeOfficeApiUtil
+    ) {
+        this.restTemplate = restTemplate;
+        this.homeOfficeSearchByParametersUrl = homeOfficeEndpoint + homeOfficeSearchByParametersUrl;
+        this.homeOfficeApiUtil = homeOfficeApiUtil;
+    }
+
+    public String getCaseStatus(AsylumCase asylumCase) {
+
+        try {
+            return restTemplate
+                .postForObject(
+                    homeOfficeSearchByParametersUrl,
+                    createRequestString(asylumCase),
+                    String.class
+                );
+
+        } catch (RestClientResponseException clientEx) {
+            throw new HomeOfficeResponseException(
+                clientEx.getResponseBodyAsString(),
+                clientEx
+            );
+
+        }
+    }
+
+    HttpEntity<String> createRequestString(AsylumCase asylumCase) {
+
+        String homeOfficeReference = (String) asylumCase.read(
+            AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER)
+            .orElse("invalid");
+
+        return new HttpEntity<>("{"
+            + "\"messageHeader\": {"
+            + "\"consumer\": {"
+            + "\"code\": \"HMCTS\","
+            + "\"description\": \"HM Courts and Tribunal Service\""
+            + "},"
+            + "\"correlationId\": \"ABC2344BCED2234EA\","
+            + "\"eventDateTime\": \"2020-05-26T17:32:28Z\""
+            + "},"
+            + "\"searchParams\": ["
+            + "{"
+            + "\"spType\": \"DOCUMENT_REFERENCE\","
+            + "\"spValue\": \""
+            + homeOfficeReference
+            + "\"}"
+            + "]}", homeOfficeApiUtil.getHomeOfficeHeader());
```

Comment:
Any reason why you've created the payload like this?  I think it's ok to get working right now since we are still in dev but I think we should be modelling this request properly as your next task.

Reply:
Modified model and removed String. Removed test